src/issues/currentIssue.ts

```diff
@@ -183,75 +198,124 @@ export class CurrentIssue extends Disposable {
+
+			if (
+				branch.name?.toLowerCase().includes(issueNumberStr.toLowerCase()) ||
+				branch.name?.toLowerCase().includes(issueTitle.replace(/[^a-zA-Z0-9-]/g, '-').toLowerCase()) ||
+				branch.name?.toLowerCase().includes(issueTitle.replace(/[^a-zA-Z0-9-]/g, '_').toLowerCase())
+			) {
+				branchesWithIssueMatch.push(branchItem);
+			} else {
+				otherBranches.push(branchItem);
 			}
+		});
+
+		// Create QuickPick items
+		const branchItems: vscode.QuickPickItem[] = [
+			{ label: 'Suggested branch', kind: vscode.QuickPickItemKind.Separator },
+			{
+				label: `$(lightbulb) ${suggestedBranchName}`,
+				description: '',
+				detail: 'Recommended branch name based on settings',
+				picked: true,
+			},
+		];
+
+		if (branchesWithIssueMatch.length > 0) {
+			branchItems.push({ label: 'Branches matching', kind: vscode.QuickPickItemKind.Separator });
+			branchItems.push(...branchesWithIssueMatch);
 		}
-		if (!this._branchName) {
-			this._branchName = await vscode.window.showInputBox({
-				value: branchNameConfig,
-				prompt: vscode.l10n.t('Enter the label for the new branch.'),
-			});
+
+		if (otherBranches.length > 0) {
+			branchItems.push({ label: 'Other branches', kind: vscode.QuickPickItemKind.Separator });
+			branchItems.push(...otherBranches);
 		}
-		if (!this._branchName) {
-			// user has cancelled
+
+		branchItems.push(
+			{ label: 'Custom branch name:', kind: vscode.QuickPickItemKind.Separator },
+			{ label: '$(pencil) Enter a custom branch name...', description: 'Choose this to type your own branch name' }
+		);
+
+		// Show QuickPick for branch selection
+		const quickPick = vscode.window.createQuickPick<vscode.QuickPickItem>();
+		quickPick.items = branchItems;
+		quickPick.placeholder = 'Select a branch or create a new one for this issue';
+		quickPick.ignoreFocusOut = true;
+		quickPick.activeItems = [branchItems[1]];
+		quickPick.show();
+
+		const selectedBranch = await new Promise<vscode.QuickPickItem | undefined>((resolve) => {
+			quickPick.onDidAccept(() => {
+				resolve(quickPick.selectedItems[0]);
+				quickPick.hide();
+			});
+			quickPick.onDidHide(() => resolve(undefined));
+		});
+
+		quickPick.dispose();
+
+		if (!selectedBranch) {
+			if (!silent) vscode.window.showInformationMessage('Branch selection cancelled.');
 			return false;
 		}
 
+		if (selectedBranch.label === '$(pencil) Enter a custom branch name...') {
+			const customBranchName = await vscode.window.showInputBox({
+				prompt: 'Enter your custom branch name',
+				placeHolder: 'e.g., feature/my-custom-branch',
+				value: suggestedBranchName,
+				validateInput: (input) => (input.trim() === '' ? 'Branch name cannot be empty.' : undefined),
+			});
+
+			if (!customBranchName) {
+				if (!silent) vscode.window.showInformationMessage('Branch creation cancelled.');
+				return false;
+			}
+
+			this._branchName = customBranchName.trim();
+		} else {
+			this._branchName = selectedBranch.label.replace(/^\$\([^\)]+\)\s*/, '').trim();
```

Comment:
Instead of futzing with the label after the branch has been picked, you can extend `QuickPickItem` and add a new property to it to track the actual branch name associated with the quick pick item.